sqlite.go

```diff
@@ -81,16 +81,14 @@ func (dialector Dialector) ClauseBuilders() map[string]clause.ClauseBuilder {
 		},
 		"LIMIT": func(c clause.Clause, builder clause.Builder) {
 			if limit, ok := c.Expression.(clause.Limit); ok {
-				if limit.Limit > 0 {
-					builder.WriteString("LIMIT ")
-					builder.WriteString(strconv.Itoa(limit.Limit))
+				if limit.Limit > 0 || limit.Offset > 0 {
+					if limit.Limit <= 0 {
+						limit.Limit = -1
+					}
+					builder.WriteString("LIMIT " + strconv.Itoa(limit.Limit) + " ")
```

Comment:
Should remove the last " " and add a prefix " " before the OFFSET?

Reply:
Did it. But personally I prefer to insert whitespaces from only one side of the expression, be it left always or right always.

Reply:
But if you are not using OFFSET, it will add a useless whitespace